[docs] Update migration guide for stable release (#8092)
mui/mui-x: docs/data/migration/migration-data-grid-v5/migration-data-grid-v5.md

<p class="description">This guide describes the changes needed to migrate the Data Grid from v5 to v6.</p>

## Start using the new release

In `package.json`, change the version of the data grid package to `latest` or `^6.0.0`.

```diff
-"@mui/x-data-grid": "5.X.X",

> +"@mui/x-data-grid": "latest",

Reviewer: ```suggestion
+"@mui/x-data-grid": "^6.0.0",
```

I think this would be preferred for most of the users to avoid accidental upgrade to v7 when it's released

```

> Using `latest` ensures that it will always use the latest v6 release, but you can also use a fixed version, like `6.0.0`.

Reviewer: > Using `latest` ensures that it will always use the latest v6 release

Not 100% correct - it will use v7 once it's released.
We can remove this statement completely as it doesn't seem to bring any additional value.

> Since v6 is a major release, it contains some changes that affect the public API.

Reviewer 2: ```suggestion
Since v6 is a major release, it contains changes that affect the public API.
```

These changes were done for consistency, improved stability and to make room for new features.

> Below are described the steps you need to take to migrate from v5 to v6.

Reviewer 2: ```suggestion
Described below are the steps needed to migrate from v5 to v6.
```
